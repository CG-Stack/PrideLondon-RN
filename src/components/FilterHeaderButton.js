// @flow
import React from "react";
import { StyleSheet } from "react-native";
import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
import Text from "./Text";
import Touchable from "./Touchable";
import { filterButtonTextColor } from "../constants/colors";

type Props = {
  text: string,
  label: string,
  onPress: Function,
  onRef?: Function,
  style?: StyleObj
};

const FilterHeaderButton = ({ text, label, onPress, onRef, style }: Props) => (
  <Touchable
    style={[styles.button, style]}
    onPress={onPress}
    ref={onRef}
    accessibilityLabel={label}
  >
    <Text style={styles.buttonText}>{text}</Text>
  </Touchable>
);

FilterHeaderButton.defaultProps = {
  onRef: undefined,
  style: {}
};

const styles = StyleSheet.create({
  button: {
    paddingHorizontal: 8,
    alignItems: "center",
    justifyContent: "center"
  },
  buttonText: {
    color: filterButtonTextColor,
    fontFamily: "Roboto-Medium"
  }
});

export default FilterHeaderButton;
