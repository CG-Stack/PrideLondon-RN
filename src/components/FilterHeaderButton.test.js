// @flow
import React from "react";
import { shallow } from "enzyme";
import FilterHeaderButton from "./FilterHeaderButton";

it("renders correctly", () => {
  const style = { marginTop: 16 };
  const output = shallow(
    <FilterHeaderButton
      text="1-2-3 filtered"
      style={style}
      onPress={() => {}}
    />
  );
  expect(output).toMatchSnapshot();
});

it("renders badge value when provided", () => {
  const style = { marginTop: 16 };
  const output = shallow(
    <FilterHeaderButton
      text="1-2-3 filtered"
      style={style}
      onPress={() => {}}
      badgeValue={2}
    />
  );
  expect(output).toMatchSnapshot();
});
