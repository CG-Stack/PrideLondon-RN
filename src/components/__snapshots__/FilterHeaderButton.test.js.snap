// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  accessibilityComponentType="button"
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "rgb(44, 218, 157)",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 32,
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      Object {
        "marginTop": 16,
      },
    ]
  }
>
  <Text
    markdown={false}
    style={
      Object {
        "color": "rgb(44, 218, 157)",
      }
    }
    type="text"
  >
    1-2-3 filtered
  </Text>
</TouchableOpacity>
`;
